src/runtime_src/core/common/debug_ip.cpp

```diff
@@ -0,0 +1,299 @@
+/**
+ * Copyright (C) 2021-2022 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#define XRT_CORE_COMMON_SOURCE
+
+#include "debug_ip.h"
+
+#include "core/include/xcl_perfmon_parameters.h"
+
+namespace xrt_core { namespace debug_ip {
+
+std::vector<uint64_t> getAIMCounterResult(const xrt_core::device* device, debug_ip_data* dbgIpData)
```

Comment:
One minor request.  For consistency prefer` lower_case `functions and variables in all core/common/*.
Also, please keep everything to the left, e.g.
```
return_type
function_name(params...)
{
}
```
And lastly, please add comments.